app/services/users.js

```diff
@@ -0,0 +1,18 @@
+const { ValidationError } = require('sequelize');
+const { hash } = require('bcrypt');
+
+const { User } = require('../models');
+const { sequelizeValidationError } = require('../errors');
+
+const withHashedPassword = ({ password, ...userParams }) => {
+  const saltRounds = 10;
+  return hash(password, saltRounds).then(hashedPassword => ({ ...userParams, password: hashedPassword }));
+};
+
+exports.createUser = userParams =>
+  withHashedPassword(userParams)
+    .then(user => User.create(user))
+    .catch(error => {
+      if (error instanceof ValidationError) throw sequelizeValidationError(error);
```

Comment:
I think this error is for validation (like trying to put a an integer in `first_name`). Correct me if I'm wrong. If that is so, this validations should be in middlewares, because it's a request validation.

Reply:
No, this catch the error when email is not unique